llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -1908,6 +1908,13 @@ InstructionCost RISCVTTIImpl::getArithmeticInstrCost(
     return BaseT::getArithmeticInstrCost(Opcode, Ty, CostKind, Op1Info, Op2Info,
                                          Args, CxtI);
 
+  // f16 with zvfhmin and bf16 will be promoted to f32.
```

Comment:
Do we need to cost the extend and truncate?

Reply:
My thinking was that given a sequence of [b]f16 ops we're probably only going to need to extend it and truncate it once. So adding that to the cost of every single arithmetic instruction seems like it would overestimate it too much.

Reply:
We extend and truncate around every op. We can't fold them away as it would change the rounding.